### ISSUE TITLE
[Callout] Heading truncated when scrolling due to incorrect bounding box in small viewports

### ISSUE THREAD

**Issue by statm** [microsoft/fluentui#19025]
### Environment Information

- **Package version(s)**: v8

### Please provide a reproduction of the bug in a codepen:
https://codepen.io/statm/pen/WNjjpjW

This is generated from the [Callout's doc page](https://developer.microsoft.com/en-us/fluentui#/controls/web/callout), with only more text content added.

#### Actual behavior:

When browser window is small enough (or zoomed in) to trigger callout content scrolling, the callout's heading is truncated:
![Fluent incorrect callout bounding](https://user-images.githubusercontent.com/1220327/126383460-da1bb352-8fdd-4740-aed4-8101ceb5155e.gif)

#### Expected behavior:

The heading and the upper border of callout should be visible.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: OOUI

**Comment by ecraig12345** [comment]
I think this is an issue with the styling applied to the callout content, not an issue with Callout itself. One workaround would be apply styles similar to this ([full demo](https://codesandbox.io/s/19025-callout-overflow-sxtbx?file=/src/index.tsx)):
```ts
const calloutStyles: Partial<ICalloutContentStyles> = {
  calloutMain: {
    padding: '20px 24px',
    boxSizing: 'border-box', // count padding as part of the height
    maxWidth: 320, // fix horizontal overflow
    maxHeight: '100%' // fix vertical overflow
  }
};
```

Ideally the examples on the website should be updated to handle this better too, but that will be low priority.

**Comment by tanisha2710** [comment]
@ecraig12345  I would like to help on this could I get more info on this

**Comment by ecraig12345** [comment]
@tanisha2710 Thanks! Here's what I'd suggest. Sorry if this is unnecessarily prescriptive but I figured that's better than not enough detail, since I don't know what your background is.

While looking at the code, I noticed there are already some top-level props for width and height (I didn't know about these when writing my earlier comment):

https://github.com/microsoft/fluentui/blob/4be0412e53a7563614bd6303ab1242dc3bc27368/packages/react/src/components/Callout/Callout.types.ts#L49-L65

However the types and doc comments are a bit wrong: in CSS you can also specify width as a string, so that should be added to the types. Also the part that mentions 0 is wrong, since the actual code doesn't have a default.

If you could update each of those three props similar to the following, that would be great:
```ts
  /**
   * Custom width for callout including borders. If not specified, no width is applied.
   */
  calloutWidth?: number | string;
```

For the examples, there are several different Callout example files, which are all pretty similar.

In each file's `styles.callout`, remove the `width` and add `boxSizing: 'border-box'` (based on my testing in codesandbox, this is needed to make the sizing work properly when the callout content has padding).
https://github.com/microsoft/fluentui/blob/4be0412e53a7563614bd6303ab1242dc3bc27368/packages/react-examples/src/react/Callout/Callout.Basic.Example.tsx#L51-L54

Then when rendering the Callout itself, add the max width and height props, and ideally a comment explaining why they're there:
```tsx
<Callout
  // ... other props as before ...
  // Using a max width and height prevents the callout from overflowing on small or highly zoomed screens
  calloutMaxWidth={320}
  calloutMaxHeight="100%"
  />
```

For setting up your environment if you haven't already, we have [this wiki page](https://github.com/microsoft/fluentui/wiki/Setup) with instructions. Once your general setup is done, this is what you'll need to do to build and update some generated files:
- `yarn` - install dependencies
- `yarn build --to @fluentui/react-examples` - build up to the package containing the examples. This is needed to help validate your change, and to update the API report file for `@fluentui/react` since you're changing a public API.
- `yarn workspace @fluentui/react update-snapshots` - some of our tests take snapshots of the components' DOM, so this command updates them (I forget whether Callout has any snapshots, so if this does nothing it's fine)
- Commit the changes
- Run `yarn change` to generate a change file (type should be `minor` because you changed a public API)--this tells our automated release build how to bump the package version based on your change, and sets the message that goes in the change log

**Comment by smhigley** [comment]
Fixed in #19472
